### PR TITLE
Prompts Desktop App to ask for network setting before connecting first time

### LINKED ISSUE #540: Electron should present Network Settings before connecting first time
---
Linux.  Using 2.3.5

Install Zulip, join an organization, enter email and password.  Zulip connects.

Instead you should be presented Network options.  Depending on your situation it could be a security problem to connect *without* setting up your proxy first.

### PR DESCRIPTION
**Pull request by muskankhedia**
---

**What's this PR do?**
Fixes #540 
This PR avails a network settings option to the user in the add organization screen before he/she connects to an organization for the first time.

**Any background context you want to provide?**
I have created and linked an option `Network and Proxy Settings >>` in the add new organization screen with the network settings in the settings menu. So now the user will be given an option to set up his network settings before connecting the first time which will redirect him directly to the network settings page. 

**Screenshots?**
**Edit 1:** A demo of the working of the functionality.
ezgif com-video-to-gif

**You have tested this PR on:**
  - [ ] Windows
  - [x] Linux/Ubuntu
  - [ ] macOS

### CODE REVIEW

**Comment by akashnimare** [comment]
@muskankhedia hey thanks for working on this. Few things -
* Please discuss about the implementations/logic/UI etc in the [desktop](https://chat.zulip.org/#narrow/stream/16-desktop) channel before sending a PR. This saves a lot of time.
* We don't need the `startintray` thing since `startminimized`is already there. So please remove those changes.
* Please go through the contributing guidelines [here](https://github.com/zulip/zulip-desktop/blob/405d938223570ffa9658bd609ba46c1f30c86b63/CONTRIBUTING.md). Also, look into other commits for commit messages etc.
* Check out our [GitHub guide](https://zulip.readthedocs.io/en/latest/git-guide.html) and [commit guidelines](https://zulip.readthedocs.io/en/latest/contributing/version-control.html).

**Comment by muskankhedia** [comment]
Yaa sure. I will keep a note of all of them.

**Comment by akashnimare** [comment]
Awesome. Also, go through #540. I think for the link, we should remove `>>` and use font-awesome icon -

image


We are using this in the `Settings > Organizations page` so you can just copy-paste that code.

**Comment by muskankhedia** [comment]
@akashnimare I have removed the `startintray` function and the linking and had modified the UI as it is discussed in #540. Please have a look and suggest if any further changes required.

**Comment by akashnimare** [comment]
image


Please align the link icon properly and add bottom margin to the button.

**Comment by muskankhedia** [comment]
@akashnimare Please have a look at the design and suggest if any other changes required.

Screenshot from 2019-09-25 19-48-59

**Comment by akashnimare** [comment]
Looks good.

File: app/renderer/js/pages/preference/new-server-form.ts [hunk 1]
```diff
@@ -30,15 +30,21 @@ class NewServerForm extends BaseComponent {
...
+					<div class="server-save-action">
+						<button id="open-create-org-link">${t.__('Create a new organization')}</button>
+					</div>
 				</div>
 				<div class="server-center">
-				<div class="server-save-action">
-					<button id="open-create-org-link">${t.__('Create a new organization')}</button>
-			</div>
+					<div class="server-network-option">
+						<span id="open-network-settings"> Network and Proxy Settings</span>
```
Comment (akashnimare) [inline app/renderer/js/pages/preference/new-server-form.ts]: You need to add the translate function here. `$t.('')}`

**Review by akashnimare — CHANGES_REQUESTED** [review]

File: app/renderer/js/pages/preference/new-server-form.ts (hunk 1 above)
Comment (muskankhedia) [reply app/renderer/js/pages/preference/new-server-form.ts]: Done !

**Comment by muskankhedia** [comment]
I have resolved the changes. I have come up with a new UI adding style for focus and hover on the `Network and Proxy Settings` button. Pls have a look on the [link](https://drive.google.com/file/d/19uBd-TyPj_mdBByuDMY4Krkt54LHWhnu/view). And suggest the one which is better I will implement it.

**Comment by akashnimare** [comment]
First one looks good I think.

**Comment by muskankhedia** [comment]
I am removing some extra unrequired lines from my commits and making it clear to be mergeable.

**Comment by akashnimare** [comment]
@muskankhedia can you squash all the commits into one so that I can merge this PR?

File: app/translations/en-US.json [hunk 2]
```diff
@@ -104,5 +104,13 @@
 	"Edit Shortcuts": "Edit Shortcuts",
 	"View Shortcuts": "View Shortcuts",
 	"History Shortcuts": "History Shortcuts",
-	"Window Shortcuts": "Window Shortcuts"
+	"Window Shortcuts": "Window Shortcuts",
+	"Add a Zulip organization": "Add a Zulip organization",
```
Comment (kanishk98) [inline app/translations/en-US.json]: I think you committed this file by mistake. Would you mind unstaging this?

File: app/renderer/js/main.ts [hunk 3]
```diff
@@ -994,6 +994,10 @@ class ServerManagerView {
 				webview.send('set-idle');
 			});
 		});
+
+		ipcRenderer.on('open-network-settings', () => {
+			this.openSettings('Network');
```
Comment (kanishk98) [inline app/renderer/js/main.ts:999]: I think you'll need to take care that the back button isn't enabled around here. This is what I get in the settings page when I click on your button:

image

Try playing around with `canGoBackButton()` in `webview.ts` and either enabling that button to go back to the new server form or disabling it entirely. I'd prefer the first option. @akashnimare what do you think?

**Review by kanishk98 — COMMENTED** [review]
Looks good, but we need to fix the back button and then squash the bug fix commits.
Feel free to DM me on czo if you find `git rebase` confusing (everyone does at first)

**Comment by muskankhedia** [comment]
> Looks good, but we need to fix the back button and then squash the bug fix commits.
> Feel free to DM me on czo if you find `git rebase` confusing (everyone does at first)

Sure

File: app/translations/en-US.json (hunk 2 above)
Comment (muskankhedia) [reply app/translations/en-US.json]: Yaa I will be unstaging it.

File: app/renderer/js/main.ts (hunk 3 above)
Comment (muskankhedia) [reply app/renderer/js/main.ts:999]: @kanishk98 I have tried for the different possibilities with the back button which are present in the following [link](https://drive.google.com/file/d/1Yo4Aytco1DccpxvnLCOBtWbGUYaOpdLf/view). However, I feel the back button behaves in a buggy manner as it starts working only after reloading the page. I feel the initLeftSidebarEvents() activates all the sidebar keys on initiating the app. So we don't have to activate or link the sidebar buttons again in every function. Please correct me if I am wrong.

**Comment by akashnimare** [comment]
@muskankhedia let me know if it's done.

File: app/renderer/js/main.ts (hunk 3 above)
Comment (muskankhedia) [reply app/renderer/js/main.ts:999]: @akashnimare @kanishk98 Pls have a look and correct me if I am wrong.

**Comment by muskankhedia** [comment]
> @muskankhedia let me know if it's done.

@akashnimare I have squashed the commits. Please have a look.

**Comment by akashnimare** [comment]
You can check out the older commit messages for an example -

https://github.com/zulip/zulip-desktop/commits/master

**Comment by muskankhedia** [comment]
> You can check out the older commit messages for an example -
> 
> https://github.com/zulip/zulip-desktop/commits/master

@akashnimare Please have a look. Does the recent commit message follow the guidelines?

**Comment by akashnimare** [comment]
> network: Prompts Desktop App to ask for network setting in add-org page (#821)

Should be -

> network: Prompts Desktop App to ask for network setting in add-org page.

Commit message -
> Fixes: #540.

**Comment by akashnimare** [comment]
I have updated the commit message and merged. Thanks, @muskankhedia.

**Pull request merged**